src/unix/qnx.c

```diff
@@ -0,0 +1,104 @@
+/* Copyright libuv contributors. All rights reserved. 
+  * 
+  * Permission is hereby granted, free of charge, to any person obtaining a copy 
+  * of this software and associated documentation files (the "Software"), to 
+  * deal in the Software without restriction, including without limitation the 
+  * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or 
+  * sell copies of the Software, and to permit persons to whom the Software is 
+  * furnished to do so, subject to the following conditions: 
+  * 
+  * The above copyright notice and this permission notice shall be included in 
+  * all copies or substantial portions of the Software. 
+  * 
+  * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+  * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+  * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+  * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+  * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS 
+  * IN THE SOFTWARE. 
+  */ 
+
+#include "uv.h"
+#include "internal.h"
+
+#include <sys/stat.h>
+
+void uv_loadavg(double avg[3]) {
+
+  avg[0] = 0.0;
+  avg[1] = 0.0;
+  avg[2] = 0.0;
+}
+
+
+int uv_exepath(char* buffer, size_t* size) {
+  ssize_t n;
+  int fd = -1;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return UV_EINVAL;
+
+  n = *size - 1;
+  if (n > 0) {
+    fd = open("/proc/self/exefile", O_RDONLY);
+    if (fd < 0)
+      return UV__ERR(errno); 
+  }
+  
+  n = read(fd, buffer, n);
```

Comment:
Should probably retry on `EINTR`? I added `uv__slurp()` recently, feel free to move that to a common file and reuse it:
https://github.com/libuv/libuv/blob/ff3ab317a2082fb450818b892156840af5e0cc35/src/unix/linux-core.c#L970-L993
Note also the use of `uv__close_nocheckstdio()`.